ephyviewer/datasource/epochs.py

```diff
@@ -54,152 +55,219 @@ class WritableEpochSource(InMemoryEpochSource):
     epoch is dict
     { 'time':np.array, 'duration':np.array, 'label':np.array, 'name':' ''}
     """
-    def __init__(self, epoch, possible_labels, color_labels=None):
-        InMemoryEpochSource.__init__(self, all_epochs=[epoch])
+    def __init__(self, epoch=None, possible_labels=[], color_labels=None, channel_name=''):
         
-        #~ self._t_stop = max([ np.max(e['time']+e['duration']) for e in self.all if len(e['time'])>0])
+        self.possible_labels = possible_labels
+        self.channel_name = channel_name
```

Comment:
What is trhis **channel_name** for ?
Is it something else than self.all[0]['name'] ?

Reply:
No, `channel_name` is simply used to set `self.all[0]['name']`. In your original example of `CsvEpochSource`, you used `'name': 'animal_state'`, which was ultimately accessible to `BaseEventAndEpoch.get_channel_name`. The new `channel_name` parameter allows the user to set this value.